docs/freqai.md

```diff
@@ -0,0 +1,265 @@
+
+Freqai is a module designed to automate a variety of tasks associated with
+training a regressor to predict signals based on input features. Among the
+the features includes:
+
+* Easy large feature set construction based on simple user input
+* Sweep model training and backtesting to simulate consistent model retraining through time
+* Smart outlier removal of data points from prediction sets using a Dissimilarity Index.
+* Data dimensionality reduction with Principal Component Analysis
+* Automatic file management for storage of models to be reused during live
+* Smart and safe data standardization
+* Cleaning of NaNs from the data set before training and prediction.
+
+TODO:
+* live is not automated, still some architectural work to be done
+
+## Background and vocabulary
+
+**Features** are the quantities with which a model is trained. $X_i$ represents the
+vector of all features for a single candle. In Freqai, the user
+builds the features from anything they can construct in the strategy.
+
+**Labels** are the target values with which the weights inside a model are trained
+toward. Each set of features is associated with a single label, which is also
+defined within the strategy by the user. These labels look forward into the
+future, and are not available to the model during dryrun/live/backtesting.
+
+**Training** refers to the process of feeding individual feature sets into the
+model with associated labels with the goal of matching input feature sets to
+associated labels.
+
+**Train data** is a subset of the historic data which is fed to the model during
+training to adjust weights. This data directly influences weight connections
+in the model.
+
+**Test data** is a subset of the historic data which is used to evaluate the
+intermediate performance of the model during training. This data does not
+directly influence nodal weights within the model.
+
+## Configuring the bot
+### Example config file
+The user interface is isolated to the typical config file. A typical Freqai
+config setup includes:
+
+```json
+    "freqai": {
+                "timeframes" : ["5m","15m","4h"],
+                "full_timerange" : "20211220-20220220",
+                "train_period" : "month",
+                "backtest_period" : "week",
+                "identifier" :  "unique-id",
+                "base_features": [
+                        "rsi",
+                        "mfi",
+                        "roc",
+                ],
+                "corr_pairlist": [
+                        "ETH/USD",
+                        "LINK/USD",
+                        "BNB/USD"
+                ],
+                "train_params" : {
+                        "period": 24,
+                        "shift": 2,
+                        "drop_features": false,
+                        "DI_threshold": 1,
+                        "weight_factor":  0,
+                },
+                "SPLIT_PARAMS" : {
```

Comment:
`SPLIT_PARAMS` and `CLASSIFIER_PARAMS` have been renamed, and should be reflected in the doc.